File: react-java/backend/src/main/java/com/clt/hrm/core/administration/role/dtos/RoleAuthDto.java
package com.clt.hrm.core.administration.role.dtos;

import com.clt.hrm.infra.common.dtos.BaseDto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class RoleAuthDto extends BaseDto {
	/**
	 * 
	 */
	private static final long serialVersionUID = 7033103627046578138L;
	private String roleId;
    private String pgmId;
    private String permId;
    
    //Optional
    private String pgmCd;
    private String pgmTpCd;
    private String permCd;
    private String activeYn;
}

File: nuxtjs-nestjs/backend/src/apps/administration/roles/dto/role-auth.dto.ts
import { IsBoolean, IsOptional, IsString } from 'class-validator';
import { BaseDto } from '@infra/common/dto';

export class RoleAuthDto extends BaseDto {
  @IsOptional()
  @IsString()
  roleId?: string;

  @IsOptional()
  @IsString()
  pgmId?: string;

  @IsOptional()
  @IsString()
  permId?: string;

  @IsOptional()
  @IsString()
  pgmCd?: string;

  @IsOptional()
  @IsString()
  pgmTpCd?: string;

  @IsOptional()
  @IsString()
  permCd?: string;
}
